app/controllers/dev_controller.rb

```diff
@@ -20,61 +22,76 @@ def reset
+
+      log_destroy(ConsentNotification.where(session: sessions))
+      log_destroy(SessionNotification.where(session: sessions))
+      log_destroy(VaccinationRecord.where(session: sessions))
 
       patient_sessions = PatientSession.where(session: sessions)
-      GillickAssessment.where(patient_session: patient_sessions).destroy_all
-      PreScreening.where(patient_session: patient_sessions).destroy_all
-      patient_sessions.destroy_all
+      log_destroy(GillickAssessment.where(patient_session: patient_sessions))
+      log_destroy(PreScreening.where(patient_session: patient_sessions))
+      patient_sessions.in_batches { log_destroy(it) }
 
-      sessions.destroy_all
+      log_destroy(sessions)
 
       patients = organisation.patients
 
-      SchoolMove.where(patient: patients).destroy_all
-      SchoolMove.where(organisation:).destroy_all
-      SchoolMoveLogEntry.where(patient: patients).destroy_all
-      AccessLogEntry.where(patient: patients).destroy_all
-      NotifyLogEntry.where(patient: patients).destroy_all
-      VaccinationRecord.where(patient: patients).destroy_all
+      log_destroy(SchoolMove.where(patient: patients))
+      log_destroy(SchoolMove.where(organisation:))
+      log_destroy(SchoolMoveLogEntry.where(patient: patients))
+      log_destroy(AccessLogEntry.where(patient: patients))
+      log_destroy(NotifyLogEntry.where(patient: patients))
+      # In local dev we can end up with NotifyLogEntries without a patient
+      log_destroy(NotifyLogEntry.where(patient_id: nil))
+      log_destroy(VaccinationRecord.where(patient: patients))
 
-      ConsentForm.where(organisation:).destroy_all
-      Consent.where(organisation:).destroy_all
-      Triage.where(organisation:).destroy_all
+      log_destroy(ConsentForm.where(organisation:))
+      log_destroy(Consent.where(organisation:))
+      log_destroy(Triage.where(organisation:))
 
-      patients.includes(:parents).destroy_all
+      patients.includes(:parents).in_batches { log_destroy(it) }
 
       batches = Batch.where(organisation:)
-      VaccinationRecord.where(batch: batches).destroy_all
-      batches.destroy_all
+      log_destroy(VaccinationRecord.where(batch: batches))
+      log_destroy(batches)
 
-      VaccinationRecord.where(
-        performed_ods_code: organisation.ods_code
-      ).destroy_all
+      log_destroy(
+        VaccinationRecord.where(performed_ods_code: organisation.ods_code)
+      )
 
       UnscheduledSessionsFactory.new.call
     end
 
-    head :ok
+    response.stream.write "Done"
+  ensure
+    response.stream.close
   end
 
   def random_consent_form
     Faker::Config.locale = "en-GB"
 
-    session = Session.includes(programmes: :vaccines).find(params[:session_id])
+    session =
+      if params[:slug].present?
+        Session.includes(programmes: :vaccines).find_by(slug: params[:slug])
+      else
+        Session.includes(programmes: :vaccines).find(slug: params[:session_id])
+      end
```

Comment:
Do we need to include this change here?

Reply:
Probably not. I just found it easier to use, do you think it's worth removing?